File: src/RomantoInteger/RomanToInteger.cpp
package RomantoInteger;

#include <arrays>
#include <string>
#include <cctype>
#include <map>
#include <iostream>

using namespace std;
 class RomanToInteger {
    public:
     static int romanToInt(string s) {
        int value = 0;
        Map<Char, int> symbolValues;
        symbolValues['I'] = 1;
        symbolValues['V'] = 5;
        symbolValues['X'] = 10;
        symbolValues['L'] = 50;
        symbolValues['C'] = 100;
        symbolValues['D'] = 500;
        symbolValues['M'] = 1000;

        //convert the string to upper case
        for (int i = 0; i < s.size(); i++) {
           s[i] = toupper(s[i]);
        }

        //process input
        for (int i = 0; i < s.size(); i++) {
            char symbol = s.charAt(i);
            switch (symbol) {
                case 'I':
                    if (i + 1 != s.size() && (s[i + 1] == 'V' || s[i + 1] == 'X')) {
                        value += symbolValues[s[i + 1]] - 1;
                        i++;
                    } else {
                        value += symbolValues[symbol];
                    }
                    break;
                case 'X':
                    if (i + 1 != s.size() && (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C')) {
                        value += symbolValues[s[i + 1]] - symbolValues[symbol];
                        i++;
                    } else {
                        value += symbolValues[symbol];
                    }
                    break;
                case 'C':
                    if (i + 1 != s.size() && (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M')) {
                        value += symbolValues[s[i + 1]] - symbolValues[symbol];
                        i++;
                    } else {
                        value += symbolValues[symbol];
                    }
                    break;
                case 'V':
                case 'L':
                case 'D':
                case 'M':
                    value += symbolValues[symbol];
                    break;
                default:
                    break;
            }
        }
        return value;
    }


    int main(String[] args) {
        cout << romanToInt("CDD"));
        return 0;
    }

}

File: src/RomantoInteger/RomanToInteger.java
package RomantoInteger;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class RomanToInteger {
    public static int romanToInt(String s) {
        int value = 0;
        Map<Character, Integer> symbolValues = new HashMap<Character, Integer>();
        symbolValues.put('I', 1);
        symbolValues.put('V', 5);
        symbolValues.put('X', 10);
        symbolValues.put('L', 50);
        symbolValues.put('C', 100);
        symbolValues.put('D', 500);
        symbolValues.put('M', 1000);
        s = s.toUpperCase();

        //process input
        for (int i = 0; i < s.length(); i++) {
            char symbol = s.charAt(i);
            switch (symbol) {
                case 'I':
                    if (i + 1 != s.length() && (s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X')) {
                        value += symbolValues.get(s.charAt(i + 1)) - 1;
                        i++;
                    } else {
                        value += symbolValues.get(symbol);
                    }
                    break;
                case 'X':
                    if (i + 1 != s.length() && (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C')) {
                        value += symbolValues.get(s.charAt(i + 1)) - symbolValues.get(symbol);
                        i++;
                    } else {
                        value += symbolValues.get(symbol);
                    }
                    break;
                case 'C':
                    if (i + 1 != s.length() && (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M')) {
                        value += symbolValues.get(s.charAt(i + 1)) - symbolValues.get(symbol);
                        i++;
                    } else {
                        value += symbolValues.get(symbol);
                    }
                    break;
                case 'V':
                case 'L':
                case 'D':
                case 'M':
                    value += symbolValues.get(symbol);
                    break;
                default:
                    break;
            }
        }
        return value;
    }


    public static void main(String[] args) {
        System.out.println(romanToInt("CDD"));//"MCMXCIV"));
    }

}

